Added flags and tests for createDecoder(). #24

emilelm96 opened the pull request:
Now there are flags and coverage checking for createDecoder().

emilelm96 committed:
Added flags and tests. #24

Now there are flags and coverage checking for createDecoder().

emilelm96 committed:
Moved coverage output. #24

Now coverage output is more easily accessible via grep.

apeinot commented on src/main/java/com/jsoniter/extra/GsonCompatibilityMode.java:
```diff
@@ -342,100 +345,127 @@ public Object decode(JsonIterator iter) throws IOException {
                         String input = iter.readString();
```
I think you should add flag into the try

apeinot reviewed (changes requested):
I think you should add two more flags:
- one into the try
- one before the last return of the function

francislaus commented:
> I think you should add two more flags:
> 
>     * one into the try
> 
>     * one before the last return of the function

I think there should be no flag in the try scope as it is not a branch itself. It gets executed in all cases

emilelm96 commented:
Could we get some confirmation on this, @Samuelhertzberg or @francislaus?

francislaus reviewed (approved):
After further review, I think that it is good as it currently is

Samuelhertzberg commented:
What about `iter.readBoolean() ? "true" : "false";` ?

emilelm96 committed:
Flagged conditional assignment and end of function. #24

As per request, I now flag the conditional assignment in createDecoder() and the last line before the functions returns.

apeinot69 committed:
Merge branch 'coverage' into issue24

Resolve conflicts before merging into coverage
This fix #24
